File: ciam/src/main/java/com/crm/ciam/security/BenutzerDetails.java
package com.crm.ciam.security;

import java.util.Collection;
import java.util.stream.Collectors;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import com.crm.ciam.entity.Benutzer;

public class BenutzerDetails implements UserDetails {

    private final Benutzer benutzer;

    public BenutzerDetails(Benutzer benutzer) {
        this.benutzer = benutzer;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return benutzer.getRollen().stream()
                .map(rolle -> new SimpleGrantedAuthority("ROLE_" + rolle.name()))
                .collect(Collectors.toSet());
    }

    @Override
    public String getPassword() {
        return benutzer.getPasswort();
    }

    @Override
    public String getUsername() {
        return benutzer.getBenutzername();
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return benutzer.isAktiv();
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return benutzer.isAktiv();
    }

    public Benutzer getBenutzer() {
        return benutzer;
    }
}

File: ciam/src/main/kotlin/com/crm/ciam/security/BenutzerDetails.kt
package com.crm.ciam.security

import com.crm.ciam.entity.Benutzer
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

class BenutzerDetails(val benutzer: Benutzer) : UserDetails {

    override fun getAuthorities(): Collection<GrantedAuthority> =
        benutzer.rollen.map { rolle -> SimpleGrantedAuthority("ROLE_${rolle.name}") }.toSet()

    override fun getPassword(): String = benutzer.passwort

    override fun getUsername(): String = benutzer.benutzername

    override fun isAccountNonExpired(): Boolean = true

    override fun isAccountNonLocked(): Boolean = benutzer.aktiv

    override fun isCredentialsNonExpired(): Boolean = true

    override fun isEnabled(): Boolean = benutzer.aktiv
}
